### ISSUE TITLE
[HW Accel Support]: for VBOX host on MacPro Quad-Core Xeon

### ISSUE THREAD

**Issue by FS1961** [blakeblackshear/frigate#5016]
### Describe the problem you are having

No frames received, check error logs in UI.

I'm sure the hw-accl is wrong. Please help

### Version

0.12.0-0DBF909

### Frigate config file

```yaml
mqtt:
  enabled: True
  host: 192.168.178.82
  user: mqtt
  password: haosmqtt

cameras:
  Dining: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t@192.168.178.59:554/play1.sdp # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
    motion:
      mask:
        - 85,207,134,279,174,352,106,352,0,352,0,104,0,0,86,0
        - 394,36,389,229,313,188,315,42

  Garage: # <------ Name the camera
    motion:
      mask:
        - 110,352,91,220,173,227,351,352
        - 578,352,402,352,473,158,563,167
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t@192.168.178.36:554/play1.sdp          # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution

  Front-door: # <------ Name the camera
    motion:
      mask:
        - 0,127,0,352,226,352,242,327,206,83
        - 640,352,640,218,506,318,356,352,240,331,227,352
    zones:
      entry_doors:
        coordinates: 293,318,513,302,574,63,285,27
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t@192.168.178.28:554/play1.sdp # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution

  Garten-house: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t@192.168.178.46:554/play1.sdp # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution
    motion:
      mask:
        - 0,480,241,480,171,403,0,100
        - 640,480,572,480,511,399,560,252,640,168
        
  Drive-way: # <------ Name the camera
    motion:
      mask:
        - 206,0,123,176,195,360,0,360,0,0
        - 640,180,640,0,530,0,451,0
    zones:
      driveway_area:
        coordinates: 469,117,596,360,226,360,224,115
      front_entrance:
        coordinates: 224,112,463,105,442,0,211,0
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t22!@192.168.178.55:554/h264  # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 2560 # <---- update for your camera's resolution
      height: 1440 # <---- update for your camera's resolution

  Patio: # <------ Name the camera.
    motion:
      mask:
        - 640,0,436,0,397,120,640,129
        - 575,126,521,221,488,196,525,129
        - 0,283,0,0,273,0,198,54,151,65,179,220,125,273
    ffmpeg:
      inputs:
        - path: rtsp://frigate:spyb0t22!@192.168.178.74:554/h264  # <----- Update for your camera
          roles:
            - detect
            - restream
    detect:
      width: 2560 # <---- update for your camera's resolution
      height: 1440 # <---- update for your camera's resolution

detectors:
  coral:
    type: edgetpu
    device: usb
  
ffmpeg:
  hwaccel_args: -c:v h264_cuvid
```

### docker-compose file or Docker CLI command

```yaml
none
```

### Relevant log output

```shell
2023-01-11 08:38:50.776051243  [2023-01-11 09:38:50] frigate.video                  ERROR   : Dining: ffmpeg process is not running. exiting capture thread...
2023-01-11 08:38:54.289917028  [2023-01-11 09:38:54] asyncio                        ERROR   : Task exception was never retrieved
2023-01-11 08:38:54.289929260  future: <Task finished name='Task-2' coro=<set_gpu_stats() done, defined at /opt/frigate/frigate/stats.py:114> exception=FileNotFoundError(2, 'No such file or directory')>
2023-01-11 08:38:54.289931391  Traceback (most recent call last):
2023-01-11 08:38:54.289933683    File "/opt/frigate/frigate/stats.py", line 146, in set_gpu_stats
2023-01-11 08:38:54.289951782      nvidia_usage = get_nvidia_gpu_stats()
2023-01-11 08:38:54.289954465    File "/opt/frigate/frigate/util.py", line 862, in get_nvidia_gpu_stats
2023-01-11 08:38:54.289956872      p = sp.run(
2023-01-11 08:38:54.289958714    File "/usr/lib/python3.9/subprocess.py", line 505, in run
2023-01-11 08:38:54.289960603      with Popen(*popenargs, **kwargs) as process:
2023-01-11 08:38:54.289962435    File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
2023-01-11 08:38:54.289986229      self._execute_child(args, executable, preexec_fn, close_fds,
2023-01-11 08:38:54.289988459    File "/usr/lib/python3.9/subprocess.py", line 1823, in _execute_child
2023-01-11 08:38:54.290003634      raise child_exception_type(errno_num, err_msg, err_filename)
2023-01-11 08:38:54.290005859  FileNotFoundError: [Errno 2] No such file or directory: 'nvidia-smi'
2023-01-11 08:39:02.281817056  [2023-01-11 09:39:02] asyncio                        ERROR   : Task exception was never retrieved
2023-01-11 08:39:02.281824885  future: <Task finished name='Task-5' coro=<set_gpu_stats() done, defined at /opt/frigate/frigate/stats.py:114> exception=FileNotFoundError(2, 'No such file or directory')>
2023-01-11 08:39:02.281826942  Traceback (most recent call last):
2023-01-11 08:39:02.281828842    File "/opt/frigate/frigate/stats.py", line 146, in set_gpu_stats
2023-01-11 08:39:02.281830610      nvidia_usage = get_nvidia_gpu_stats()
2023-01-11 08:39:02.281832536    File "/opt/frigate/frigate/util.py", line 862, in get_nvidia_gpu_stats
2023-01-11 08:39:02.281834114      p = sp.run(
2023-01-11 08:39:02.281835948    File "/usr/lib/python3.9/subprocess.py", line 505, in run
2023-01-11 08:39:02.281837608      with Popen(*popenargs, **kwargs) as process:
2023-01-11 08:39:02.281839418    File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
2023-01-11 08:39:02.281841182      self._execute_child(args, executable, preexec_fn, close_fds,
2023-01-11 08:39:02.281842957    File "/usr/lib/python3.9/subprocess.py", line 1823, in _execute_child
2023-01-11 08:39:02.281844691      raise child_exception_type(errno_num, err_msg, err_filename)
2023-01-11 08:39:02.281846510  FileNotFoundError: [Errno 2] No such file or directory: 'nvidia-smi'
2023-01-11 08:39:06.350712381  [2023-01-11 09:39:06] watchdog.Dining                ERROR   : Ffmpeg process crashed unexpectedly for Dining.
2023-01-11 08:39:06.351099560  [2023-01-11 09:39:06] watchdog.Dining                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-01-11 08:39:06.352505448  [2023-01-11 09:39:06] ffmpeg.Dining.detect           ERROR   : [h264_cuvid @ 0x55a7cdd1de40] Cannot load libnvcuvid.so.1
2023-01-11 08:39:06.352509496  [2023-01-11 09:39:06] ffmpeg.Dining.detect           ERROR   : [h264_cuvid @ 0x55a7cdd1de40] Failed loading nvcuvid.
```

### FFprobe output from your camera

```shell
none
```

### Operating system

HassOS

### Install method

HassOS Addon

### Network connection

Wired

### Camera make and model

D-link and others

**Comment by FS1961** [comment]
UI screen:

<img width="1365" alt="Screenshot 2023-01-11 at 10 19 32 AM" src="https://user-images.githubusercontent.com/98458806/211766869-92faad31-fa4e-4ef0-a862-5f3c0c4e65b6.png">

Custom Frigate Card (HA)"
<img width="1424" alt="Screenshot 2023-01-11 at 10 20 52 AM" src="https://user-images.githubusercontent.com/98458806/211767190-f8623416-0256-45fd-a827-e63119da21eb.png">

**Comment by NickM-27** [comment]
Xeon doesn't have integrated GPU so there's no way to use hardware acceleration

**Comment by FS1961** [comment]
Okay, what about the GPU?

Chipset Model:	NVIDIA GeForce GTX 680
  Type:	GPU
  Bus:	PCIe
  Slot:	Slot-1
  PCIe Lane Width:	x16
  VRAM (Dynamic, Max):	2048 MB
  Vendor:	NVIDIA (0x10de)
  Device ID:	0x1180
  Revision ID:	0x00a2
  ROM Revision:	3731
  Metal:	Supported, feature set macOS GPUFamily1 v3

**Comment by NickM-27** [comment]
You are using home assistant OS which does not have support for nvidia drivers so nvidia GPU can not be used with the addon

> Also, I have an additional CPU assigned to the VBOX, how do I add this?

not sure what you're asking, VBOX should use it if you have assigned it to the HA OS VM

**Comment by FS1961** [comment]
Can I assign it as a detector, like this?

detectors:
  coral:
    type: edgetpu
    device: usb
  cpu1:
    type: cpu

**Comment by NickM-27** [comment]
> Can I assign it as a detector, like this?
> 
> detectors: coral: type: edgetpu device: usb cpu1: type: cpu

we 100% discourage using CPU detectors with a coral, it will hurt performance not help it due to the CPU having slow inference times

**Comment by FS1961** [comment]
Roger that.

Is there anyway to improve the performance? Right now, I'm pegging my cpu at 132%

**Comment by NickM-27** [comment]
what does your system page show? That has all the important CPU information

**Comment by FS1961** [comment]
<img width="1400" alt="Screenshot 2023-01-11 at 4 02 27 PM" src="https://user-images.githubusercontent.com/98458806/211840229-b0b20d0c-812f-4753-b38b-17e6d9c7d3f6.png">

**Comment by NickM-27** [comment]
your inference speeds are incredibly high, what USB port (2 or 3, USB C or USB A?) are you using with your coral

other than that the CPU usage looks fine for the different camera processes

**Comment by FS1961** [comment]
It's a USB3.0 USB A.

Composite Device:

  Product ID:	0x9302
  Vendor ID:	0x18d1  (Google Inc.)
  Version:	1.00
  Speed:	Up to 5 Gb/sec
  Location ID:	0x00600000 / 23
  Current Available (mA):	900
  Current Required (mA):	896
  Extra Operating Current (mA):	0

**Comment by NickM-27** [comment]
Something seems really off because coral should have inference time of 10 ms not 1000

**Comment by NickM-27** [comment]
Nothing changed with the way the coral is handled in 0.12, the inference speed is most likely some sort of hardware or upstream software problem. Do you have the HA integration setup? You could look at the graph that shows the coral inference speeds

**Comment by FS1961** [comment]
This?
<img width="1122" alt="Screenshot 2023-01-11 at 4 22 46 PM" src="https://user-images.githubusercontent.com/98458806/211845076-14faf76c-4943-444d-8a08-e63765c2810f.png">

**Comment by NickM-27** [comment]
Oh, it looks like the inference speed entity is not enabled so there is no way to know what it was before.

**Comment by FS1961** [comment]
I was just going to restore my whole HA system to a previous backup. Doesn't that work?

**Comment by NickM-27** [comment]
Yes it should

**Comment by FS1961** [comment]
Okay, I'll do that.

**Comment by FS1961** [comment]
Back at 11.1 and here is the data. Looks like 1/4?

<img width="405" alt="Screenshot 2023-01-11 at 5 10 30 PM" src="https://user-images.githubusercontent.com/98458806/211857287-d0738930-de63-4300-9ce3-d7e82d108074.png">

**Comment by NickM-27** [comment]
Hmm, it's probably bouncing around, still way higher than a coral should be. Might be a virtualbox problem with the USB bus or maybe you have other USB devices that are using bandwidth.

**Comment by FS1961** [comment]
I'll check that and get back to you on this thread tomorrow.

**Comment by FS1961** [comment]
NickM-27 - I played around with the system quite a bit this morning. I've tried going back to a previous version but it seems the CPU utilization always remains high. I've deleted the Frigate directories and uninstalled the add-on. I'm hoping to start from scratch again at this point. I've also moved the Coral to a different port/bus. Is there a way to run this on a separate VBox and if so, what OS should I use?

Without Frigate, my VBox utilization is:
<img width="674" alt="Screenshot 2023-01-12 at 11 09 03 AM" src="https://user-images.githubusercontent.com/98458806/212038845-ab160076-ce57-4ece-b8fa-b1d50c18252c.png">

running Frigate, it just pegs 100% pretty much.

**Comment by NickM-27** [comment]
we recommend running in debian, it's native debian that is recommended but a VM still may perform better than the addon inside HA OS

**Comment by FS1961** [comment]
Thanks, how about a hardware suggestion. I have 6 cameras but planning on 2 more.

**Comment by NickM-27** [comment]
https://deploy-preview-4055--frigate-docs.netlify.app/frigate/hardware/

**Comment by FS1961** [comment]
Thanks, I've seen this list. I'm not sure what "several" means.

**Comment by FS1961** [comment]
Okay, so I restored back to 11.1 and this is what I'm getting. 

I'm not sure what the upgrade is doing but the Coral seems much lower at 11.1

<img width="408" alt="Screenshot 2023-01-13 at 8 04 10 AM" src="https://user-images.githubusercontent.com/98458806/212258081-dd125f2c-ca71-4785-b92f-c71e21c09418.png">

**Comment by NickM-27** [comment]
Corals should be at 10 ms so something still seems really off is my point. Nothing changed with 0.12 in regards to how the coral runs detections.

**Comment by FS1961** [comment]
I think it does run at 10ms when it's not detecting. See below.
 I am going to try and stand up another VM 
<img width="407" alt="Screenshot 2023-01-12 at 8 20 45 AM" src="https://user-images.githubusercontent.com/98458806/212365507-adad576f-c957-434b-9bad-50d8683fa8d0.png">
<img width="404" alt="Screenshot 2023-01-12 at 7 44 04 AM" src="https://user-images.githubusercontent.com/98458806/212365579-3859acb7-a9c8-4beb-8787-9294fd4e7258.png">

**Comment by NickM-27** [comment]
No, the default value frigate starts with is 10 ms and then once the detector starts frigate begins the average. So it showing exactly 10 just means nothing has been run through detection yet

**Comment by FS1961** [comment]
Hello NickM-27 - I've built a Debian barebones VM and am running. Running much better. I think I still have some tuning to do but here is what the system is reporting now. Much better improvement and my HA is running @ 2%.
<img width="1016" alt="Screen Shot 2023-01-14 at 7 51 30 PM" src="https://user-images.githubusercontent.com/98458806/212491318-c0cc6789-4914-4e8f-8e7d-3428da96ed05.png">

**Comment by NickM-27** [comment]
Yeah that does look to be an improvement

**Comment by FS1961** [comment]
This seems to be a good solution for me so I'm closing this issue.

**Comment by FS1961** [comment]
@NickM-27 - just wanted to close the loop with you. I purchased a small N5105 computer and installed Debian server standalone with HA Proxy. I do have one question though, how do I get the hardware acceleration to work? The unit is an 11 gen Intel Celeron N5105 with Intel UHD 605 GPU.

I've tried but it seems to blow up. Any ideas?

ffmpeg:
  hwaccel_args: -c:v h264_qsv


Here is the difference on the inference:

<img width="1537" alt="Screenshot 2023-02-13 at 7 06 12 AM" src="https://user-images.githubusercontent.com/98458806/218383103-4bbd7588-f9c6-4c9e-8a95-da4835c75fe2.png">
